community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/codegen/CodeStructure.scala

```diff
@@ -155,7 +156,11 @@ trait MethodStructure[E] {
 object CodeStructure {
   val __TODO__MOVE_IMPLEMENTATION = new CodeStructure[GeneratedQueryExecution] {
     override def generateQuery(packageName: String, className: String, columns: Seq[String], opIds: Seq[String])(block: MethodStructure[_] => Unit) = {
-      val generator = codegen.CodeGenerator.generateCode(CodeStructure.getClass.getClassLoader/*, SourceCode.PRINT_SOURCE*/)
+      val generator = try {
+        codegen.CodeGenerator.generateCode(CodeStructure.getClass.getClassLoader/*, SourceCode.PRINT_SOURCE*/)
+      } catch {
+        case e: Exception => throw new CantCompileQueryException(e.getMessage, e)
```

Comment:
Does this mean that we'll fallback to interpreted runtime on any error and not only on `CodeGenerationNotSupportedException`?